komorebi/utilities.py

```diff
@@ -0,0 +1,237 @@
+		wallpaper = VideoWallpaper(screen, wallpaper_config)
+	elif wallpaper_type == WallpaperType.WEB.value:
+		from komorebi.wallpapers.web import WebWallpaper
+		wallpaper = WebWallpaper(screen, wallpaper_config)
+	else:
+		logging.warning(f"Invalid wallpaper type: {wallpaper_type}")
+		raise RuntimeError("Invalid wallpaper type")
+
+	return wallpaper
+
+def load_overlays(screen, wallpaper_config: GLib.KeyFile) -> List[Overlay]:
+	overlays = []
+
+	for overlay in wallpaper_config.get_string('Info', 'Order').split(','):
+		if overlay == OverlayType.CLOCK.value:
+			from komorebi.overlays.clock import Clock
+			overlays.append(Clock(screen, wallpaper_config))
+		if overlay == OverlayType.ASSET.value:
+			from komorebi.overlays.asset import Asset
+			overlays.append(Asset(screen, wallpaper_config))
+		# elif overlay == OverlayType.DESKTOP.value: 	# This doesn't exist on wallpaper specification *yet*
+	if komorebi.settings.show_desktop_icons:					# so for now we simply use this
+		from komorebi.overlays.desktop import Desktop
+		overlays.append(Desktop(screen))
+
+	return overlays
+
+def on_settings_changed(screen, setting_key : komorebi.settings.keys) -> List[Overlay]:
+	overlays = []
+
+	if setting_key == komorebi.settings.keys.SHOW_DESKTOP \
+	and komorebi.settings.show_desktop_icons:
+		from komorebi.overlays.desktop import Desktop
+		overlays.append(Desktop(screen))
+
+	return overlays
+
+# Returns an icon detected from file, IconTheme, etc...
+def get_icon_from(icon: str, icon_size: int) -> GdkPixbuf.Pixbuf:
+	iconPixbuf : GdkPixbuf.Pixbuf = None
+
+	if icon == None or icon == '':
+		return iconPixbuf
+
+	# Try those methods:
+	# 1- Icon is a file, somewhere in '/'.
+	# 2- Icon is an icon in a IconTheme.
+	# 3- Icon isn't in the current IconTheme.
+	# 4- Icon is not available, use default.
+	if Gio.File.new_for_path(icon).query_exists():
+		iconPixbuf = GdkPixbuf.Pixbuf.new_from_file_at_scale(icon, icon_size, icon_size, False)
+		return iconPixbuf
+
+	iconTheme : Gtk.IconTheme = Gtk.IconTheme.get_default()
+	iconTheme.prepend_search_path('/usr/share/pixmaps')
+
+	try:
+		iconPixbuf = iconTheme.load_icon(icon, icon_size, Gtk.IconLookupFlags.FORCE_SIZE)
+	except GLib.Error:
+		if iconPixbuf == None:
+			iconPixbuf = iconTheme.load_icon('application-default-icon', icon_size, Gtk.IconLookupFlags.FORCE_SIZE)
+
+	return iconPixbuf
+
+def apply_css(widgets, css):
+	provider = Gtk.CssProvider()
+	provider.load_from_data(str.encode(css))
+
+	for widget in widgets:
+		widget.get_style_context().add_provider(provider, 800)
+
+def apply_alpha(widgets):
+	for widget in widgets:
+		widget.set_visual(widget.get_screen().get_rgba_visual() or widget.get_screen().get_system_visual())
+
+# A dirty way to check if gstreamer is installed
+def can_play_videos():
+	# FIXME: Horrible way to detect presence of libgstlibav.so
+	paths = [
+		'/usr/lib',
+		'/usr/lib64',
+		'/usr/lib/i386-linux-gnu',
+		'/usr/lib/x86_64-linux-gnu',
+		'/usr/lib/arm-linux-gnueabihf'
```

Comment:
We probably also want the `/usr/local` and `~/.local` variants? Just a FYI, can be done in a separate PR.

Reply:
Also be careful with the loading order @Ev1lbl0w . `~/.local` should override `/usr/local` and the same for `/usr` mimicking `$PATH` behavior.

Reply:
Honestly, this piece of code is just *begging* to be replaced; this is no proper way to detect if a package/`*.so` file exists. I didn't go forward with it already because I don't want to introduce many features/bugfixes on what's mostly intended to be just a major code refactoring.

Reply:
Actually...

```
$ pkgconf --variable=pluginsdir gstreamer-1.0
/usr/lib/gstreamer-1.0
```